app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -159,6 +159,10 @@ def save_hosts_inventory(ems, hashes, target = nil)
           #   and the new hostname is not an ip address
           h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)
 
+          # Update the name to be found.name if it has the same ems_ref and the name
+          # already has a '- int' suffix to work around duplicate hostnames
+          h[:name] = found.name if found.name =~ /#{h[:name]} - \d+$/ && h[:ems_ref] == found.ems_ref
```

Comment:
Can you collapse this conditional with the one above since they both affect the `h[:name]`?

Also, what is the case where the ems_ref would *not* match in this case?  Would it be if we take the host from another EMS?  If so, wouldn't that be bad if we *don't* change the name?